EIPS/eip-4824.md

```diff
@@ -0,0 +1,220 @@
+---
+eip: 4824
+title: DAO API
+description: An API standard for decentralized autonomous organizations (DAOs).
+author: Joshua Tan (@thelastjosh), Isaac Patka (@ipatka), Ido Gershtein <ido@daostack.io>, Eyal Eithcowich <eyal@deepdao.io>, Michael Zargham (@mzargham), Sam Furter (@nivida)
+discussions-to: https://ethereum-magicians.org/t/eip-4824-decentralized-autonomous-organizations/8362
+status: Draft
+type: Standards Track
+category: ERC
+created: 2022-02-17
+---
+
+## Abstract
+
+An API standard for decentralized autonomous organizations (DAOs), focused on relating on-chain and off-chain representations of membership and proposals.
+
+## Motivation
+
+DAOs, since being invoked in the Ethereum whitepaper, have been vaguely defined. This has led to a wide range of patterns but little standardization or interoperability between the frameworks and tools that have emerged. Standardization and interoperability are necessary to support a variety of use-cases. In particular, a standard daoURI, similar to tokenURI in [ERC-721](./eip-721), will enhance DAO discoverability, legibility, proposal simulation, and interoperability between tools. More consistent data across the ecosystem is also a prerequisite for future DAO standards.
+
+## Specification
+
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+Every EIP-4824 compliant contract MUST implement the EIP4824 interface below:
+
+```solidity
+pragma solidity ^0.4.20;
+
+/// @title EIP-4824 DAOs
+/// @dev See https://eips.ethereum.org/EIPS/eip-4824
+
+interface EIP4824 {
+    /// @notice A distinct Uniform Resource Identifier (URI) pointing to a JSON object following the "EIP-4824 DAO JSON-LD Schema". This JSON file splits into four URIs: membersURI, proposalsURI, activityLogURI, and governanceURI. The membersURI should point to a JSON file that conforms to the "EIP-4824 Members JSON-LD Schema". The proposalsURI should point to a JSON file that conforms to the "EIP-4824 Proposals JSON-LD Schema". The activityLogURI should point to a JSON file that conforms to the "EIP-4824 Activity Log JSON-LD Schema". The governanceURI should point to a flatfile, normatively a .md file. Each of the JSON files named above can be statically-hosted or dynamically-generated.
+    function daoURI() external view returns (string _daoURI);
+}
+```
+
+The EIP-4824 DAO JSON-LD Schema mentioned above.
+
+```json
+{
+	"@context": "http://www.daostar.org/schemas",
```

Comment:
I'm not sure what this is (and the link 404s), but would it make sense to include whatever is supposed to be there in the `assets` directory of this EIP instead?

Reply:
Looks like the redirect hasn't been set up; that should get fixed this week! 

I think keeping a snapshot of the context in the assets folder is a good idea. That said, the context files / the ontology will need to grow & change with additional API extensions built on top of this standard.

Reply:
> I think keeping a snapshot of the context in the assets folder is a good idea. That said, the context files / the ontology will need to grow & change with additional API extensions built on top of this standard.

Those extensions can provide their own schema in their own `assets` directory.

Reply:
Added the current schema to the assets directory.

Reply:
Can we remove`http://www.daostar.org/` per https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1.md#linking-to-external-resources?